mne/time_frequency/tfr.py

```diff
@@ -1137,156 +1434,151 @@ def save(self, fname, overwrite=False):
+        if n_times != len(times):
+            raise ValueError("Number of times and data size don't match"
+                             " (%d != %d)." % (n_times, len(times)))
+        self.data = data
+        self.times = np.array(times)
+        self.freqs = np.array(freqs)
+        self.comment = comment
+        self.method = method
 
+    def __repr__(self):
+        s = "time : [%f, %f]" % (self.times[0], self.times[-1])
+        s += ", freq : [%f, %f]" % (self.freqs[0], self.freqs[-1])
+        s += ", epochs : %d" % self.data.shape[0]
+        s += ', channels : %d' % self.data.shape[1]
+        return "<EpochsTFR  |  %s>" % s
 
-def read_tfrs(fname, condition=None):
-    """
-    Read TFR datasets from hdf5 file.
+    def average(self):
+        data = np.mean(self.data, axis=0)
+        return AverageTFR(info=self.info.copy(), data=data,
+                          times=self.times.copy(), freqs=self.freqs.copy(),
+                          nave=self.data.shape[0],
+                          method=self.method)
+
+
+def combine_tfr(all_tfr, weights='nave'):
+    """Merge AverageTFR data by weighted addition.
+
+    Create a new AverageTFR instance, using a combination of the supplied
+    instances as its data. By default, the mean (weighted by trials) is used.
+    Subtraction can be performed by passing negative weights (e.g., [1, -1]).
+    Data must have the same channels and the same time instants.
 
     Parameters
     ----------
-    fname : string
-        The file name, which should end with -tfr.h5 .
-    condition : int or str | list of int or str | None
-        The condition to load. If None, all conditions will be returned.
-        Defaults to None.
-
-    See Also
-    --------
-    write_tfrs
+    all_tfr : list of AverageTFR
+        The tfr datasets.
+    weights : list of float | str
+        The weights to apply to the data of each AverageTFR instance.
+        Can also be ``'nave'`` to weight according to tfr.nave,
+        or ``'equal'`` to use equal weighting (each weighted as ``1/N``).
 
     Returns
     -------
-    tfrs : list of instances of AverageTFR | instance of AverageTFR
-        Depending on `condition` either the TFR object or a list of multiple
-        TFR objects.
+    tfr : AverageTFR
+        The new TFR data.
 
     Notes
     -----
-    .. versionadded:: 0.9.0
+    .. versionadded:: 0.11.0
     """
+    tfr = all_tfr[0].copy()
+    if isinstance(weights, string_types):
+        if weights not in ('nave', 'equal'):
+            raise ValueError('Weights must be a list of float, or "nave" or '
+                             '"equal"')
+        if weights == 'nave':
+            weights = np.array([e.nave for e in all_tfr], float)
+            weights /= weights.sum()
+        else:  # == 'equal'
+            weights = [1. / len(all_tfr)] * len(all_tfr)
+    weights = np.array(weights, float)
+    if weights.ndim != 1 or weights.size != len(all_tfr):
```

Comment:
Could some of these compatibility checks be put into a `_check_compat` function that would be called both for this case as well as for the various arithmetic functions defined above? (e.g., for `__sub__` it's just checking if times / freqs are the same, but not channels etc...a lot of that checking is done here so it could be a separate utility function to be used any time 2+ TFR objects are being merged)